components/uniqueness/backing-store-impl/src/integrationTest/kotlin/net/corda/uniqueness/backingstore/impl/JPABackingStoreImplIntegrationTests.kt

```diff
@@ -0,0 +1,683 @@
+                                )
+                            )
+                        )
+                    }
+                }
+            }
+
+            // Persisting this error should throw because its size if bigger than 1024.
+            val invalidErrorMessage = "e".repeat(maxErrMsgLength + 1)
+            assertThrows<IllegalStateException> {
+                backingStoreImpl.session(aliceIdentity) { session ->
+                    session.executeTransaction { _, txnOps ->
+                        txnOps.commitTransactions(
+                            listOf(
+                                Pair(
+                                    internalRequest, UniquenessCheckResultFailureImpl(
+                                        now(), UniquenessCheckErrorMalformedRequestImpl(invalidErrorMessage)
+                                    )
+                                )
+                            )
+                        )
+                    }
+                }
+            }
+        }
+
+        @Test
+        fun `Persisting rejected transaction with general error succeeds`() {
+            val txId = SecureHashUtils.randomSecureHash()
+            val txIds = listOf(txId)
+            val errorMessage = "some error message"
+            val txns = listOf(
+                Pair(
+                    generateRequestInternal(txId), UniquenessCheckResultFailureImpl(
+                        now(), UniquenessCheckErrorMalformedRequestImpl(errorMessage)
+                    )
+                )
+            )
+
+            backingStoreImpl.session(aliceIdentity) { session ->
+                session.executeTransaction { _, txnOps -> txnOps.commitTransactions(txns) }
+            }
+
+            val txnDetails = mutableMapOf<SecureHash, UniquenessCheckTransactionDetailsInternal>()
+            backingStoreImpl.session(aliceIdentity) { session ->
+                txnDetails.putAll(session.getTransactionDetails(txIds))
+            }
+
+            UniquenessAssertions.assertContainingTxId(txnDetails = txnDetails, txIds = txIds)
+            UniquenessAssertions.assertMalformedRequestResult(errorMessage, txnDetails.entries.single().value.result)
+        }
+
+        @Test
+        fun `Persisting unconsumed states succeeds`() {
+            val hashCnt = 3
+            val secureHashes = List(hashCnt) { SecureHashUtils.randomSecureHash() }
+            val stateRefs = secureHashes.map { UniquenessCheckStateRefImpl(it, 0) }
+
+            backingStoreImpl.session(aliceIdentity) { session ->
+                session.executeTransaction { _, txnOps -> txnOps.createUnconsumedStates(stateRefs) }
+            }
+
+            val stateDetails = mutableMapOf<UniquenessCheckStateRef, UniquenessCheckStateDetails>()
+            backingStoreImpl.session(aliceIdentity) { session ->
+                stateDetails.putAll(session.getStateDetails(stateRefs))
+            }
+
+            assertThat(stateDetails.size).isEqualTo(hashCnt)
```

Comment:
```suggestion
            assertThat(stateDetails.size).isEqualTo(secureHashes.size)
```

Would mean we can get rid of `hashCnt`

Reply:
Updated